script/generate-documentation

```diff
@@ -96,21 +93,51 @@ SSH_REPO=${REPO/https:\/\/github.com\//git@github.com:}
 TARGET_BRANCH="gh-pages"
 REPO_DIRECTORY=$PWD
 : ${GH_PUBLISH:=0}
+SSH_KEY=$1
+SSH_IV=$2
 
-if [ -z "${asciidoctor_bin}" -o ! -f "${asciidoctor_bin}" ]; then
+function call_asciidoctor {
+
+    if [ -z "${asciidoctor_bin}" -o ! -f "${asciidoctor_bin}" ]; then
         echo "Could not find asciidoctor binary in your path, please install it and run this command again"
         exit 1
-fi
+    fi
 
-cd docs
-mkdir ${tmpwd}/output 2>/dev/null || true # we don't care if the directory already exists
-cp -r images ${tmpwd}/output
+    cd docs
+    mkdir ${tmpwd}/output 2>/dev/null || true # we don't care if the directory already exists
+    cp -r images ${tmpwd}/output
 
-${asciidoctor_bin} -r asciidoctor-pdf -b pdf -o ${tmpwd}/output/openqa-documentation-${verbose_doc_name}.pdf index.asciidoc -d book
-${asciidoctor_bin} -o ${tmpwd}/output/openqa-documentation-${verbose_doc_name}.html index.asciidoc -d book
+    ${asciidoctor_bin} -r asciidoctor-pdf -b pdf -o ${tmpwd}/output/openqa-documentation-${verbose_doc_name}.pdf index.asciidoc -d book
+    ${asciidoctor_bin} -o ${tmpwd}/output/openqa-documentation-${verbose_doc_name}.html index.asciidoc -d book
 
-echo -e  "${green}The output has been generated at ${tmpwd}/output"
+    echo -e  "${green}The output has been generated at ${tmpwd}/output"
+
+    cd ..
+}
 
 if [ ${GH_PUBLISH} ] && [ ${CONTINUOUS_INTEGRATION} ]; then
-        update_docs
-fi;
+    if [ "${TRAVIS_BRANCH}" != "master" ]; then
+	echo "Branch is: ${TRAVIS_BRANCH}, not generating any documentation"
+	exit 0
+    fi
+
+    if [ "${TRAVIS_PULL_REQUEST}" != false ]; then
+	echo "Build is a pull request, not generating any documentation"
+	exit 0
+    fi
+
+    if [ -z "${SSH_KEY}" -o -z "${SSH_IV}" ]; then
+	echo "Build is a pull request, not generating any documentation"
+	exit 0
+    fi
+
+    echo "Requirements met, generating documentation"
```

Comment:
i'd put this into a function on its own... pretty much like [this](https://github.com/os-autoinst/openQA/compare/master...foursixnine:fix/travis?expand=1#diff-bad955a0bad4220b6c7f7b81e2704268)